### PR TITLE
test/part: improve testing for partitioned communication

### PR DESCRIPTION
**Pull request by thomasgillis**
## Pull Request Description

- add strided datatype testing a commonly used data access pattern
- add testing with prime numbers of partitions to detect aggregation issues

## Author Checklist
* [ ] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [ ] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [ ] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: test/mpi/part/Makefile.am [hunk 1]
```diff
@@ -19,6 +19,7 @@ noinst_PROGRAMS =  \
     parrived                       \
     parrived_wait                  \
     parrived_pready_out_of_order   \
+	parrived_strided 			   \
```
Comment (hzhou) [inline test/mpi/part/Makefile.am:22]: No tabs

**Review by hzhou — COMMENTED** [review]
This looks fine. Will approve once tests are clear.

**Review by raffenet — APPROVED** [review]
LGTM

**Comment by raffenet** [comment]
> LGTM

Oops, I forgot about the tab issue noted previous. @thomasgillis if you fix that up, this should be OK to merge.

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Pull request merged**